source/MaterialXCore/Definition.cpp

```diff
@@ -32,6 +33,31 @@ const string UnitDef::UNITTYPE_ATTRIBUTE = "unittype";
 // NodeDef methods
 //
 
+const string& NodeDef::getType() const
+{
+    const vector<OutputPtr>& activeOutputs = getActiveOutputs();
+
+    // Some organizational nodes have no output types
+    if (getNodeGroup() == ORGANIZATION_NODE_GROUP && activeOutputs.empty())
```

Comment:
This is the main change to handle "organization" nodedefs

Reply:
In the version of stdlib_defs.mtlx that I have all the organization nodes has outputs already. The "backdrop" node has an output of type "none". And the "dot" nodes have many different output types.

Reply:
minor: if it's an organizational node group then getting the activeOutputs is a pointless operation. Maybe the two conditions should be separated to avoid needing to get the activeOutputs if the node is organizational?

Reply:
Sorry I didn't see you removed the "backdrop" output above. I wonder though if we should keep it, even though it's technically never used, just to avoid these special cases in the code. I imagine the backdrop will need special drawing in a node editor UI anyway, so it's output would never be visible.

Reply:
This is from Doug, but I do agree that it makes no sense to make this a node with inputs and outputs as it's not what it is. 

The node editor (or rather the runtime) should know it's organizational element. Regardless of whether it's an <output> with type "none" or it just returns "none" whatever consumes this will have to deal with the same information (getType() returns "none" as type). In fact if it's an <output> with type "none" there would need to be general special casing to figure what to do with "none" types. 

I would say the better solution is to remove it as a nodedef but that's not really part of this review.

Reply:
Yes an application would need to have special casing for the "none" type. But my point was that now we get special casing _both_ in the API and in the application. If we handle "none" as any other type in the API it's only the application side that needs special casing, not both sides.